src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -719,4 +734,73 @@ private function isUninitializedValueError(\Error $e): bool
         return str_starts_with($e->getMessage(), 'Typed property')
             && str_ends_with($e->getMessage(), 'must not be accessed before initialization');
     }
+
+    /**
+     * Returns all attributes with a SerializedPath annotation and the respective path.
+     */
+    private function getNestedAttributes(string $class = null): array
+    {
+        if (!$this->classMetadataFactory || null === $class || !$this->classMetadataFactory->hasMetadataFor($class)) {
+            return [];
+        }
+
+        $properties = [];
+        $serializedPaths = [];
+        $classMetadata = $this->classMetadataFactory->getMetadataFor($class);
+        foreach ($classMetadata->getAttributesMetadata() as $name => $metadata) {
+            if (!$serializedPath = $metadata->getSerializedPath()) {
+                continue;
+            }
+            $pathIdentifier = implode(',', $serializedPath);
+            if (isset($serializedPaths[$pathIdentifier])) {
+                throw new LogicException(sprintf('Duplicate serialized path: "%s" used for properties "%s" and "%s".', $pathIdentifier, $serializedPaths[$pathIdentifier], $name));
+            }
+            $serializedPaths[$pathIdentifier] = $name;
+            $properties[$name] = $serializedPath;
+        }
+
+        return $properties;
+    }
+
+    private function removeNestedValue(array $path, array $data): array
+    {
+        $element = array_shift($path);
+        if ([] === $path) {
+            unset($data[$element]);
+        } else {
+            $data[$element] = $this->removeNestedValue($path, $data[$element]);
+            if ([] === $data[$element]) {
+                unset($data[$element]);
+            }
+        }
+
+        return $data;
+    }
+
+    private function extractNestedValue(array $path, array $data): ?string
+    {
+        foreach ($path as $element) {
+            if (!isset($data[$element])) {
+                return null;
+            }
+            $data = $data[$element];
+        }
+
+        return $data;
+    }
+
+    private function setNestedValue(array $data, array $elements, mixed $value): array
```

Comment:
Are a little though, I was wondering on relying on the `PropertyAccess` component, as it is already required by the `Serializer` component and can do the very same thing (https://symfony.com/doc/current/components/property_access.html#reading-from-arrays, https://symfony.com/doc/current/components/property_access.html#writing-to-arrays)

In that way, instead of givin an array to specify the pass, we could be able to only give a valid property access path.

WDYT?

Reply:
Thanks for the review @mtarld and thanks for already handling most of it, @nicolas-grekas! I used the the `PropertyAccessor` for the `setNestedValue()` and `extractNestedValue()` methods, and it worked like a charm. Basically makes those two methods unnecessary, as the resulting code can be included in the `updateData` and `denormalize` methods of the `AbstractObjectNormalizer`. The only downside of using the `PropertyAccessor` is that the array given in a `SerializedPath` annotation

```php
/**
 * @SerializedPath({"one", "two", "three"})
 */
```

needs to be converted into the string `[one][two][three]` as both the `PropertyAccessor` and the `PropertyPath` don't handle arrays as input. Using the `PropertyAccessor` to get rid of the extra functions in the `AbstractObjectNormalizer` would only make sense to me if the `SerializedPath` annotation is converted to a string that's already in the correct format

```php
/**
 * @SerializedPath("[one][two][three]")
 */
```

to avoid converting the array in the normalizer. I'm happy to do this, but let me know what you think about it first, @mtarld and @nicolas-grekas.

Reply:
Indeed, I think that it could be easier for users to directly use the standards provided by the property access component (and to not learn a new way to access array values)

Therefore I'll use a string instead of an array and to not convert anything